abra_core/src/typechecker/typechecker.rs

```diff
@@ -80,59 +81,43 @@ impl Typechecker {
         scope.bindings.insert(name.to_string(), binding);
     }
 
-    fn update_binding<F>(&mut self, name: &String, func: F)
```

Comment:
The `update_binding` and `update_type` methods were just too much code in order to achieve what I wanted. Esp since this all got streamlined quite a bit by using `Type::Reference`.